lib/uniq-by.js

```diff
@@ -0,0 +1,31 @@
+import curry from './curry'
+
+export default curry(_uniqBy)
+
+/**
+ * Returns a new list without duplicate elements. Uniqueness is determined by applying
+ * the supplied function to each list element. If the supplied function produces
+ * identical values for multiple list elements only the first element is kept.
+ * The order of result values is determined by the order they occur in the input list.
+ * {@link http://ecma-international.org/ecma-262/7.0/#sec-samevaluezero SameValueZero} is used for equality comparisons
+ *
+ * @function uniqBy
+ * @arg {Function} fn A function to produce the value used to determine uniqueness
+ * @arg {Array} list The list to inspect
+ * @return {Array} A new list of unique elements
+ */
+function _uniqBy(fn, list) {
+  const set = new Set()
+  const out = []
+  let val
```

Comment:
why declare here? can be done in the loop and as `const` right?

Reply:
yes, will fix